### PR TITLE
Fixes given by pylint checker

### PR DESCRIPTION
**Pull request by pgdr**
Using pylint on ert gives some errors.  Here's a bunch of fixes that pylint discovered.

First commit is declaring vector functions in template class.

Second consists of several small fixes, changes in variable names, changed use of deprecated-but-now-removed functions.

pylint is not good with monkey patching, nor Qt Signals, so expect errors where this is used.  It didn't support `super(·)` very well either.

### CODE REVIEW

**Comment by pgdr** [comment]
Here is log from pylint:
```
python/ert/ecl
************* Module ert.ecl.ecl_grid
E:1002,32: Instance of 'EclGrid' has no 'nz' member (no-member)
E:1003,36: Instance of 'EclGrid' has no 'ny' member (no-member)
E:1004,40: Instance of 'EclGrid' has no 'nx' member (no-member)
E:1119,12: Instance of 'EclGrid' has no '_fwrite_grdecl2' member (no-member)
************* Module ert.ecl.ecl_kw
E:133, 8: Instance of 'BaseCClass' has no '__private_init' member (no-member)
E:143, 8: Instance of 'BaseCClass' has no '__private_init' member (no-member)
E:1086, 4: Method should have "self" as first argument (no-self-argument)
************* Module ert.ecl.ecl_region
E:944, 4: Method has no argument (no-method-argument)
************* Module ert.ecl.ecl_rft_cell
E: 81, 8: Bad first argument 'RFTCell' given to super() (bad-super-call)
E:117, 8: Bad first argument 'RFTCell' given to super() (bad-super-call)
************* Module ert.ecl.ecl_sum
E:172,12: Instance of 'BaseCClass' has no '__private_init' member (no-member)
E:179, 8: Instance of 'BaseCClass' has no '__private_init' member (no-member)

python/ert/enkf
************* Module ert.enkf.enkf_main
E:162,15: Instance of 'EnKFMain' has no '_get_logh' member (no-member)
************* Module ert.enkf.meas_data
E: 63,22: Instance of 'MeasData' has no 'getActiveObsSize' member (no-member)

python/ert/job_queue
************* Module ert.job_queue.queue
E:190,15: Instance of 'JobQueue' has no '_' member (no-member)
E:199,14: Instance of 'JobQueue' has no '__getitem__' member (no-member)

python/ert/server
************* Module ert.server.simulation_context
E: 42, 8: Instance of 'CThreadPool' has no 'submitJob' member (no-member)

python/cwrap
************* Module cwrap.basecvalue
E: 54,15: cls.DATA_TYPE is not callable (not-callable)

python/ert_gui/ertwidgets
************* Module ert_gui.ertwidgets.analysismoduleselector
E: 31, 8: Value 'analysis_module_combo.currentIndexChanged' is unsubscriptable (unsubscriptable-object)
************* Module ert_gui.ertwidgets.caseselector
E: 21, 8: Value 'self.currentIndexChanged' is unsubscriptable (unsubscriptable-object)
************* Module ert_gui.ertwidgets.listeditbox
E: 32, 8: Value 'self._completer.activated' is unsubscriptable (unsubscriptable-object)

python/ert_gui/simulation
************* Module ert_gui.simulation.iterated_ensemble_smoother_panel
E: 36, 8: Value 'num_iterations_spinner.valueChanged' is unsubscriptable (unsubscriptable-object)

python/ert_gui/tools/plot
************* Module ert_gui.tools.plot.plot_case_selection_widget
E: 45, 8: Value 'self.__signal_mapper.mapped' is unsubscriptable (unsubscriptable-object)

tests/core/util
************* Module tests.core.util.test_cthread_pool
E: 44,12: Instance of 'CThreadPool' has no 'testFunction' member (no-member)

tests/ert/enkf
************* Module tests.ert.enkf.test_linalg
E: 20, 0: Unable to import 'ert.enkf.linalg' (import-error)
************* Module tests.ert.enkf.test_time_map
E: 30,12: No value for argument 'time' in method call (no-value-for-parameter)
```

File: python/python/ert/job_queue/queue.py [hunk 1]
```diff
@@ -187,20 +187,26 @@ def num_running( self ):
...
+        raise IndexError('index out of range, was: %d should be in [0, %d)' %
+                         (index, ls))
+
     def exists(self, index):
-        job = self.__getitem__(index)
-        if job:
-            return True
-        else:
-            return False
+        return self[index]
```
Comment (joakim-hove) [inline python/python/ert/job_queue/queue.py]: This will raise `IndexError` instead of returning `False`?

**Comment by joakim-hove** [comment]
> pylint is not good with monkey patching, nor Qt Signals, so expect errors where this is used. It didn't support super(·) very well either.

I see you have not integrated `pylint` in the build/test system - would that make sense?

File: python/python/ert/job_queue/queue.py (hunk 1 above)
Comment (pgdr) [inline python/python/ert/job_queue/queue.py]: To be fair, I don't know what the intended usage was; the `__getitem__` wasn't implemented prior to this patch!

Please submit an issue if `exists` should be used for something else than this.

**Pull request merged**